src/components/Button/styles.js

```diff
@@ -31,39 +32,45 @@ const styles = {
     return theme.color.text.white;
   },
   backgroundColor({ intent, disabled, variant }) {
-    if (disabled) return `${theme.color.element.primary} !important`;
+    if (disabled) return `${theme.color.element.inactive} !important`;
```

Comment:
<img width="461" alt="image" src="https://user-images.githubusercontent.com/512823/166686091-7fb7cce8-7be8-4c2e-8b15-269b22e8d6e8.png">

I think the text variant should have the text color grayed out when disabled instead of looking like the normal button disabled.